WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -436,57 +428,56 @@ local methods = {
             end
         end
 
-        function self.callbacks.OnDragStart(_, _)
+        function self.callbacks.OnDragStart(frame, _)
             if WeakAuras.IsImporting() or not data.parent then return end;
 
-            data.dragging = true
+            local x, y = GetCursorPosition()
+            local scale = UIParent:GetScale()
 
-            self.frame:StartMoving()
-            self.frame:SetUserPlaced(false)
+            frame.isDragged = true
+            frame:StartMoving()
+            frame:ClearAllPoints()
+            frame:SetPoint("Center", UIParent, "BOTTOMLEFT", x/scale, y/scale)
+
+            WeakAuras:SortDisplayButtons() -- find a better way to trigger a DoLayout call on the scroll frame
         end
 
-        function self.callbacks.OnDragStop(_)
-            if data.dragging then
-                self.frame:StopMovingOrSizing()
+        function self.callbacks.OnDragStop(frame)
```

Comment:
As to the visualization, there seems to be no callback for dragMove. So the way I would do that.

In OnDragStart, register a OnUpdate method on frame (But check if no other code does that, so that there aren't any conflits.) Unregister that method in OnDragStop.

In that OnUpdate handler, you need to basically figure out which position the frame would be added if the drop stopped at that moment. That's essentially the same code you already have.

Also you need a frame that represents the drop indicator, so add a method to this file: dropIndicator() which lazily creates a frame with a texture child.

In the OnUpdate handler you position the frame relative to the button by using SetPoint.

That doesn't sound to complicated to me.